### PR TITLE
Minor refactoring

### PR DESCRIPTION
**Pull request by utkarsh2102**
Hi @arnau :wave: 

Here's my take on doing some minor refactoring and most importantly, getting the CI go green again! :tada: 

Signed-off-by: Utkarsh Gupta <<utkarsh@debian.org>>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
-sudo: false
+---
 language: ruby
+cache: bundler
+before_install: gem install bundler -v 2.1.4
 rvm:
-  - 2.7.0
+  - 2.7.1
   - ruby-head
```
Comment (utkarsh2102) [inline .travis.yml:7]: I am also in favor of adding 2.5.x and 2.6.x support.
D'you think we can (or we should) do that?

Comment (utkarsh2102) [reply .travis.yml:7]: Something like:
```
rvm:
  - 2.5.8
  - 2.6.6
  - 2.7.1
  - ruby-head
```

Comment (arnau) [reply .travis.yml:7]: Sounds good to me 👍

File: .rubocop.yml [hunk 2]
```diff
@@ -2,33 +2,29 @@ require:
   - rubocop-packaging
 
 AllCops:
+  NewCops: enable
   Exclude:
     - bundle/vendors/**/*
+  TargetRubyVersion: 2.4
```
Comment (utkarsh2102) [inline .rubocop.yml:8]: I am not sure why the CI failed: https://travis-ci.org/github/arnau/ISO8601/jobs/705068359
Any clue?

**Comment by utkarsh2102** [comment]
> I ran rake and RuboCop seems to be complaining

What version of RuboCop d'you have?

File: .rubocop.yml (hunk 2 above)
Comment (utkarsh2102) [reply .rubocop.yml:8]: `Error: RuboCop found unsupported Ruby version 2.1 in `TargetRubyVersion` parameter (in vendor/bundle/ruby/2.7.0/gems/rainbow-3.0.0/.rubocop.yml). 2.1-compatible analysis was dropped after version 0.57.`
Even though the TargetRubyVersion is 2.4 :sweat_smile:

**Comment by arnau** [comment]
> > I ran rake and RuboCop seems to be complaining
> 
> What version of RuboCop d'you have?

```
# cat Gemfile.lock
  38   │     rubocop (0.85.1)
  39   │       parallel (~> 1.10)
  40   │       parser (>= 2.7.0.1)
  41   │       rainbow (>= 2.2.2, < 4.0)
  42   │       regexp_parser (>= 1.7)
  43   │       rexml
  44   │       rubocop-ast (>= 0.0.3)
  45   │       ruby-progressbar (~> 1.7)
  46   │       unicode-display_width (>= 1.4.0, < 2.0)
```

**Comment by utkarsh2102** [comment]
Okay, I'll put this in .rubocop.yml so this shouldn't be a blocker.

**Comment by arnau** [comment]
I use `nix` to isolate the environment and I had to add `rake` to it. Would you mind adding it to this PR?

```
# cat default.nix
let
  pkgs = import <nixpkgs> {};
  tooling = [
    pkgs.exa
    pkgs.xsv
    pkgs.git
    pkgs.ruby_2_7
    pkgs.rubyPackages_2_7.rake
  ];

in
  pkgs.mkShell {
    LANG = "en_GB.UTF-8";
    EDITOR="vim";
    buildInputs = tooling ++ [
    ];

    shellHook = ''
      set -o vi
      local pink='\e[1;35m'
      local yellow='\e[1;33m'
      local blue='\e[1;36m'
      local white='\e[0;37m'
      local reset='\e[0m'

      function git_branch {
        git rev-parse --abbrev-ref HEAD 2>/dev/null
      }

      export PS1="\[$pink\]nix \[$blue\]\W \[$yellow\]\$(git_branch)\[$white\] ∙ \[$reset\]"

      alias ls=exa

      source ${pkgs.git}/share/bash-completion/completions/git
    '';
  }
```

File: .rubocop.yml (hunk 2 above)
Comment (arnau) [reply .rubocop.yml]: I'm not sure either 🤔

**Comment by arnau** [comment]
Although, perhaps rake should be a dependency in the gemspec?

**Comment by utkarsh2102** [comment]
Okay, I'm going to experiment with a few changes and commits here. Shall squash the related commits in the end.
And then ping you at the end for a review.

Meanwhile if you've any idea about CI failure because of RubyTargetVersion, let me know 😪😅

**Comment by arnau** [comment]
If I'm reading it correctly, the problem comes from `rainbow`:

> Error: RuboCop found unsupported Ruby version 2.1 in `TargetRubyVersion` parameter (in vendor/bundle/ruby/2.7.0/gems/rainbow-3.0.0/.rubocop.yml). 2.1-compatible analysis was dropped after version 0.57

**Comment by utkarsh2102** [comment]
Weehoo, CI is green again!
https://travis-ci.org/github/arnau/ISO8601

The problem was that bundler was cached :smile:

**Comment by arnau** [comment]
What about this https://docs.rubocop.org/rubocop/configuration.html#setting-the-target-ruby-version ?

```yaml
AllCops:
  TargetRubyVersion: 2.5.0
```

**Comment by utkarsh2102** [comment]
So I am going to finally squash them and it should be ready in the next ~10 minutes.

**Comment by utkarsh2102** [comment]
@arnau, green and ready from my side! :tada:

**Comment by arnau** [comment]
Unless you have other things in mind, I'll be releasing v0.13.0 soon.

**Comment by utkarsh2102** [comment]
Sure, looks good for v0.13.0 :tada:

**Comment by arnau** [comment]
Fantastic! nice work, thanks :)

**Pull request merged**